src/main/java/prerthan/duke/command/Command.java

```diff
@@ -0,0 +1,65 @@
+package sharadhr.duke.command;
+
+import sharadhr.duke.exception.DukeEmptyDetailException;
+import sharadhr.duke.exception.DukeInvalidDateException;
+import sharadhr.duke.io.Output;
+import sharadhr.duke.io.Storage;
+import sharadhr.duke.task.TaskList;
+
+/**
+ * Models a possible user command to the Duke program.
+ */
+public abstract class Command
+{
+	protected CommandName commandName;
+	protected String[] argumentTokens;
+
+	/**
+	 * Creates a {@link Command}, with a {@link String}[] that contains the tokenised arguments
+	 * to the command, from the user input.
+	 *
+	 * @param argumentTokens the tokenised input from the user minus the first token
+	 */
+	public Command(String... argumentTokens)
+	{
+		this.argumentTokens = argumentTokens;
+	}
+
+	public static CommandName whichCommand(String token)
```

Comment:
Should the method's name be something clearer? For example `getCommandNameFromToken`